src/main/java/edu/ucsb/cs156/rec/controllers/RecommendationRequestController.java

```diff
@@ -0,0 +1,147 @@
+
+        recommendationRequestRepository.delete(recommendationRequest);
+
+        return genericMessage("RecommendationRequest with id %s deleted".formatted(id));
+    }
+
+    /**
+     * Any admin can update a single Recommendation Request
+     * 
+     * @param id       the id of the Recommendation Request to update
+     * @param incoming the updated Recommendation Request
+     * @return the updated Recommendation Request object
+     */
+    @Operation(summary = "An admin can update a single recommendation request")
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @PutMapping("/admin")
+    public RecommendationRequest updateRecommendationRequestAsAdmin(
+        @Parameter(name = "id") @RequestParam Long id,
+        @RequestBody @Valid RecommendationRequest incoming) {
+
+        RecommendationRequest recommendationRequest = 
+            recommendationRequestRepository
+                .findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(RecommendationRequest.class, id));
+
+        recommendationRequest.setProfessorName(incoming.getProfessorName());
+        recommendationRequest.setProfessorEmail(incoming.getProfessorEmail());
+        recommendationRequest.setRequesterName(incoming.getRequesterName());
+        recommendationRequest.setRecommendationTypes(incoming.getRecommendationTypes());
+        recommendationRequest.setDetails(incoming.getDetails());
+        recommendationRequest.setStatus(incoming.getStatus());
+        recommendationRequest.setSubmissionDate(incoming.getSubmissionDate());
+        recommendationRequest.setCompletionDate(incoming.getCompletionDate());
+
+        recommendationRequestRepository.save(recommendationRequest);
+
+        return recommendationRequest;
+    }
+
+    /**
+     * The user who posted a RecommendationRequest can update their RecommendationRequest
+     * 
+     * @param id       the id of the Recommendation Request to update
+     * @param incoming the updated Recommendation Request
+     * @return the updated Recommendation Request object
+     */
+    @Operation(summary = "User can update their RecommendationRequest")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @PutMapping("")
+    public RecommendationRequest updateRecommendationRequestAsUser( 
+        @Parameter(name = "id") @RequestParam Long id,
+        @RequestBody @Valid RecommendationRequest incoming) {
+
+        User currentUser = getCurrentUser().getUser(); 
+        RecommendationRequest recommendationRequest = 
+            recommendationRequestRepository
+                .findByIdAndUser(id, currentUser)
+                .orElseThrow(() -> new EntityNotFoundException(RecommendationRequest.class, id));
+
+        recommendationRequest.setProfessorName(incoming.getProfessorName());
+        recommendationRequest.setProfessorEmail(incoming.getProfessorEmail());
+        recommendationRequest.setRequesterName(incoming.getRequesterName());
```

Comment:
This is more of a design choice but when I made the POST routes, I used currentUser.getUser().getFullName() and set that as the value for setRequesterName, so I'm not sure we actually have to update the name because we are already validating that the current user is the one who made the request originally so we could design the form so that they aren't even able to add a requesterName

Reply:
But we could also just store the requesterName by what they send and then only validate on user_id

Reply:
Since I only allow the user to update and delete their own requests (by using the findByIdAndUser function), I made it so that they can't update their name in the request. That should be okay since (as you stated) the POST operation gets it from currentUser.getUser().getFullName(). Thanks for pointing that out!